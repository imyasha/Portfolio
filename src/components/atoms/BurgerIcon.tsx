import React from "react"
import styled from "styled-components"

const Burger = ({ toggle }: any) => {
  const StyledBurger = styled.button`
    position: absolute;
    top: 5%;
    left: 2rem;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    width: 2rem;
    height: 2rem;
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 0;
    z-index: 10;

    &:focus {
      outline: none;
    }

    div {
      width: 2rem;
      height: 0.25rem;
      background: ${toggle => (toggle ? "#0D0C1D" : "#EFFFFA")};
      border-radius: 10px;
      transition: all 0.3s linear;
      position: relative;
      transform-origin: 1px;

      :first-child {
        transform: ${toggle => (toggle ? "rotate(45deg)" : "rotate(0)")};
      }

      :nth-child(2) {
        opacity: ${toggle => (toggle ? "0" : "1")};
        transform: ${toggle => (toggle ? "translateX(20px)" : "translateX(0)")};
      }

      :nth-child(3) {
        transform: ${toggle => (toggle ? "rotate(-45deg)" : "rotate(0)")};
      }
    }
  `

  return (
    <StyledBurger open={toggle} onClick={() => toggle.toggle()}>
      <div />
      <div />
      <div />
    </StyledBurger>
  )
}

export default Burger
