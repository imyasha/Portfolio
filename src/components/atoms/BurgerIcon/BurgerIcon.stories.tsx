import { BurgerIcon, BurgerIconProps } from './BurgerIcon';

export default {
	title: 'atoms/BurgerIcon',
	component: BurgerIcon,
};

export const BurgerIconStory = (args: BurgerIconProps) => (
	<BurgerIcon {...args} />
);

BurgerIconStory.storyName = 'BurgerIcon';
BurgerIconStory.args = {};
