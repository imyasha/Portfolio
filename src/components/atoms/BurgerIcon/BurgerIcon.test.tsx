import { BurgerIcon } from '../BurgerIcon';
import { render } from '@testing-library/react';

describe('BurgerIcon', () => {
	it('renders correctly', () => {
		const { container } = render(<BurgerIcon isOpen={false} />);
		expect(container).toMatchSnapshot();
	});
});
