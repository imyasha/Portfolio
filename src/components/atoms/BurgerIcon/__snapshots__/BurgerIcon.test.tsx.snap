// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerIcon renders correctly 1`] = `
<div>
  <div
    class="burger-icon"
  >
    <span />
    <span />
    <span />
    <span />
  </div>
</div>
`;
